modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/BuildDeviceInfoPayload.java

```diff
@@ -53,53 +53,47 @@ public BuildDeviceInfoPayload(Context context){
-    		JSONObject property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_USERNAME);
-    		property.put(DEVICE_PROPERTY_VALUE, username);
-    		properties.put(property);
-    		
-    		property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_DESCRIPTION);
-    		property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceName());
-    		properties.put(property);
-    		
-    		property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_IMEI);
-    		property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceId());
-    		properties.put(property);
-    		
-    		property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_IMSI);
-    		property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getIMSINumber());
-    		properties.put(property);
-    		
-    		property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_MODEL);
-    		property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceModel());
-    		properties.put(property);
-    		
-    		property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_VENDOR);
-    		property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceManufacturer());
-    		properties.put(property);
-    		
-    		property = new JSONObject();
-    		property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_OS);
-    		property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getOsVersion());
-    		properties.put(property);
-    		
-    		result.put(DEVICE_PROPERTY_TAG, properties);
+			result.put(DEVICE_IDENTIFIER, deviceInfo.getMACAddress());
+			result.put(DEVICE_PROPERTY_NAME, deviceInfo.getDeviceName());
+			result.put(DEVICE_DESCRIPTION, deviceInfo.getDeviceName());
+			result.put(DEVICE_OWNERSHIP, type);
+			result.put(DEVICE_PROPERTY_OWNER, owner);
+
+			JSONArray properties = new JSONArray();
+			JSONObject property = new JSONObject();
+
+			property = new JSONObject();
+			property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_IMEI);
+			property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceId());
+			properties.put(property);
+
+			property = new JSONObject();
+			property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_IMSI);
+			property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getIMSINumber());
+			properties.put(property);
+
+			property = new JSONObject();
+			property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_MODEL);
+			property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceModel());
+			properties.put(property);
+
+			property = new JSONObject();
+			property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_VENDOR);
+			property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getDeviceManufacturer());
+			properties.put(property);
+
+			property = new JSONObject();
+			property.put(DEVICE_PROPERTY_NAME, DEVICE_PROPERTY_OS);
+			property.put(DEVICE_PROPERTY_VALUE, deviceInfo.getOsVersion());
+			properties.put(property);
+
+			result.put(DEVICE_PROPERTY_TAG, properties);
 		}catch(JSONException e){
```

Comment:
Fixed in Commit [https://github.com/wso2/product-mdm/commit/b0c66506d24fb5288300bec4224f1eaa58c7b426].